src/api/app/controllers/search_controller.rb

```diff
@@ -196,6 +196,19 @@ def search(what, render_all)
     items = find_items(what, predicate)
 
     matches = items.size
+    if render_all && search_results_exceed_configured_limit?(matches)
```

Comment:
Sounds backwards to me somehow. Passing around `matches` instead of something like

```
if matches > search_results_limit
....
```

Reply:
I agree it sounds somehow backwards. Unfortunately, the logic involved is not as straightforward as your snippet, and it is all abstracted in the call to the `search_results_exceed_configured_limit?` method.

Do you have a better/more readable suggestion, which performs the same amount of checks?

Reply:
This line:

```
records_to_show = limit || items.size
```
looks wrong to me. `limit` could be defined and be greater than `items.size`, but we still should assign `items.size` to `records_to_show`.

Reply:
We talked offline and in the controllers current form there isn't a way to do this that isn't weird. So let's take on some technical dept for the refactoring of this controller...